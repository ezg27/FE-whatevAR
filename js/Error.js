import React, { Component } from 'react';
import {
    Text,
    View,
    StyleSheet,
    ImageBackground,
    TouchableHighlight
 } 
 from 'react-native';

export default class Error extends Component {

constructor() {
    super()
    this._handleOnPress = this._handleOnPress.bind(this);
}

    render() {
        return (
            <ImageBackground source={require('./res/Mojito.jpg')} style={{width: '100%', height: '100%'}}>
            <View style={localStyles.outer} >
           <View style={localStyles.inner}>
            <Text style={localStyles.titleText}>Whoooops!</Text>
            <Text style={localStyles.titleText}>Something went wrong</Text>
            <TouchableHighlight style={localStyles.button}
                        underlayColor={'#68a0ff'} onPress={this._handleOnPress}>
                        <Text style={localStyles.buttonText}>Retry</Text>
                    </TouchableHighlight>
        </View>
        </View>
          </ImageBackground>
        )
    }

    _handleOnPress() {
    this.props.displayError();
    }
}

var localStyles = StyleSheet.create({
    outer : {
      flex : 1,
      flexDirection: 'row',
      alignItems:'center',
      justifyContent: 'center'
    },
    inner: {
      flex : 1,
      flexDirection: 'column',
      alignItems:'center'
    },
    titleText: {
        fontFamily: 'Avenir',
        fontSize: 25,
        color: '#000000',
        textAlignVertical: 'center',
        textAlign: 'center',
        fontWeight: 'bold',
        color: 'white'
    },
    buttonText: {
        fontFamily: 'Avenir',
        color: '#1D976C',
        textAlign: 'center',
        fontSize: 25,
        textAlignVertical: 'center'
    },
    button: {
        height: 45,
        width: 120,
        paddingTop: 5,
        paddingBottom: 5,
        marginTop: 40,
        marginBottom: 10,
        backgroundColor: 'white',
        borderRadius: 10
    },
})