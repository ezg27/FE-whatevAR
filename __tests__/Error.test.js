import React from 'react';
import { shallow } from 'enzyme';
import { Text, TouchableHighlight } from 'react-native';
import renderer from 'react-test-renderer';
import Error from '../js/Error.js';

describe('Error', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<Error />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('displays text', () => {
        const wrapper = shallow(<Error />);
        expect(wrapper.find(Text)).toHaveLength(3);
    });
    it('simulates click events', () => {
        const onButtonClick = jest.fn();
        const wrapper = shallow(<Error displayError={onButtonClick}/>);
        wrapper.find(TouchableHighlight).first().props().onPress();
        expect(onButtonClick.mock.calls.length).toBe(1);
    });
});