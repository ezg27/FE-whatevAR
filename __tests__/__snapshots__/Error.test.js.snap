// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders correctly 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../js/res/Mojito.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontFamily": "Avenir",
            "fontSize": 25,
            "fontWeight": "bold",
            "textAlign": "center",
            "textAlignVertical": "center",
          }
        }
      >
        Whoooops!
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontFamily": "Avenir",
            "fontSize": 25,
            "fontWeight": "bold",
            "textAlign": "center",
            "textAlignVertical": "center",
          }
        }
      >
        Something went wrong
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 10,
            "height": 45,
            "marginBottom": 10,
            "marginTop": 40,
            "paddingBottom": 5,
            "paddingTop": 5,
            "width": 120,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#1D976C",
              "fontFamily": "Avenir",
              "fontSize": 25,
              "textAlign": "center",
              "textAlignVertical": "center",
            }
          }
        >
          Retry
        </Text>
      </View>
    </View>
  </View>
</View>
`;
